parser/src/lexer.rs

```diff
@@ -222,7 +222,17 @@ impl<Id> Token<Id> {
     }
 }
 
-pub type SpannedToken<Id> = Spanned<Token<Id>>;
+#[derive(Copy, Clone, Ord, PartialOrd, Eq, PartialEq, Debug)]
+pub struct Span {
+    pub start: Location,
+    pub end: Location,
+}
+
+#[derive(Clone, Debug)]
+pub struct SpannedToken<Id> {
+    pub span: Span,
+    pub value: Token<Id>,
+}
```

Comment:
I guess these may get removed sometime later since the lexer should just need byte position and column position? Otherwise `pos::Span` could be generic over the location

```
pub struct Span<P> {
    pub start: P,
    pub end: P
}
```